### PR TITLE
Feature/lista studenti backend frontend

### PR DESCRIPTION
**Pull request by Razvan-Joita**

### CODE REVIEW

File: src/main/java/org/example/Main.java [hunk 1]
```diff
@@ -2,19 +2,13 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.CrossOrigin;
 
+@CrossOrigin(origins = "http://localhost:63342")
```
Comment (Piciorus) [inline src/main/java/org/example/Main.java]: do we need this?

File: src/main/java/org/example/controllers/StudentController.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Students> getStudentById(@PathVariable Integer id){
+        Students student = studentService.getStudentById(id).orElse(null);
+        return (student!=null) ? ResponseEntity.ok(student):ResponseEntity.notFound().build();
+
+    }
+
+    @PostMapping("/addStudent")
```
Comment (Piciorus) [inline src/main/java/org/example/controllers/StudentController.java:45]: this is not restful .... see this https://www.sitepoint.com/rest-api/

File: src/main/java/org/example/controllers/StudentController.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            studentService.deleteStudent(id);
+            return ResponseEntity.noContent().build();
+        } catch (EntityNotFoundException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body(e.getMessage());
+        } catch (Exception e) {
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(e.getMessage());
+        }
+    }
+
+    @GetMapping("/modificareStudent")
```
Comment (Piciorus) [inline src/main/java/org/example/controllers/StudentController.java]: also this is not restful

File: src/main/java/org/example/model/Students.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+import jakarta.persistence.*;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import lombok.NoArgsConstructor;
+import org.antlr.v4.runtime.misc.NotNull;
+
+
```
Comment (Piciorus) [inline src/main/java/org/example/model/Students.java]: reformat file

File: src/main/java/org/example/model/Students.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import java.time.LocalDate;
+
+@Entity
+@Getter
+@Setter
+@ToString
```
Comment (Piciorus) [inline src/main/java/org/example/model/Students.java:21]: do we need this annotation?

File: src/main/java/org/example/service/StudentService.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+            return true;
+        } else {
+            throw new EntityNotFoundException("Student with ID " + id + " not found");
+        }
+    }
+
+    public Students addStudent(
+            String firstName,
+            String lastName,
+            String cnp,
```
Comment (Piciorus) [inline src/main/java/org/example/service/StudentService.java:70]: can you add a dto for this ... we have too many params in function

File: src/main/java/org/example/service/StudentService.java [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+    public List<Students> getAllStudents() {
+        return studentRepository.findAll();
+    }
+
+    public Optional<Students> getStudentById(Integer id) {
+        return studentRepository.findById(id);
+    }
+
+    public Optional<Students> updateStudent(int studentId, Students updatedStudent) {
```
Comment (Piciorus) [inline src/main/java/org/example/service/StudentService.java:34]: create a dto for this updatedStudent

File: src/main/resources/application.properties [hunk 8]
```diff
@@ -0,0 +1,18 @@
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+spring.datasource.url=jdbc:mysql://localhost:3306/Students
+spring.datasource.username=root
+spring.datasource.password=Jabulani2002@
+spring.jpa.show-sql=true
+spring.jpa.hibernate.ddl-auto=update
```
Comment (Piciorus) [inline src/main/resources/application.properties:6]: try to create the connection to database with credentials which are at line 12

File: src/main/resources/static/mainHtmx.html [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        </select>
+        <label>Nivel studiu</label>
+        <select required class="selectstyle">
+            <option value="Licenta">Licenta</option>
+            <option value="Master">Master</option>
+        </select>
+        <label>Forma finantare</label>
+        <select required class="selectstyle">
+            <option value="Buget">Buget</option>
+            <option value="Taxa">Taxa</option>
```
Comment (Piciorus) [inline src/main/resources/static/mainHtmx.html]: it will be nice if you will add diacritics

File: src/main/resources/static/style.css [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+    border: 1px;
+    border-radius: 10px;
+    background: linear-gradient(to right, #3D5A80 29px, #D9D9D9 0);
+    padding-left: 40px;
+    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25)
+}
+
+.edit-form {
+    /*display: none;*/
+}
```
Comment (Piciorus) [inline src/main/resources/static/style.css:43]: ?

File: src/main/resources/static/style.css [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+    padding: 8px;
+    margin-bottom: 16px;
+    border: 1px solid #fffdfd;
+    border-radius: 5px;
+    background: white;
+}
+
+/*#student-list {*/
+/*    margin: 20px;*/
+/*    padding: 20px;*/
```
Comment (Piciorus) [inline src/main/resources/static/style.css:97]: ?

**Review by Piciorus — CHANGES_REQUESTED** [review]

**Comment by Piciorus** [comment]
please add a print screen to see how it looks like

File: src/main/java/org/example/service/StudentService.java (hunk 7 above)
Comment (Kl3igrec) [reply src/main/java/org/example/service/StudentService.java:34]: Nice to have, don't think it's necessary for our case

File: src/main/java/org/example/controllers/StudentController.java (hunk 3 above)
Comment (Kl3igrec) [reply src/main/java/org/example/controllers/StudentController.java]: This doesn't do anything either, dk what it does

**Comment by DragosPetri** [comment]
do on develop

**Pull request closed without merging**